src/state.rs

```diff
@@ -224,10 +224,7 @@ impl<EC: 'static + Debug, EA: 'static + Debug> AcmeState<EC, EA> {
             }
         };
         self.resolver.set_cert(Arc::new(cert));
-        let wait_duration = (validity[1] - (validity[1] - validity[0]) / 3 - Utc::now())
-            .max(chrono::Duration::zero())
-            .to_std()
-            .unwrap_or_default();
+        let wait_duration = (validity[1] - (validity[1] - validity[0]) / 3i32 - OffsetDateTime::now_utc()).unsigned_abs();
```

Comment:
Let's make this a bit safer and more obvious while we are at it. Something like this:
```
let [valid_from, valid_until] = validity;
let total_valid_duration = valid_until.saturating_sub(valid_from).max(Duration::ZERO);
let renew_at = valid_until.saturating_sub(total_valid_duration/3);
let wait_duration = renew_at.saturating_sub(OffsetDateTime::now_utc());
self.wait = Some(Timer::after(wait_duration.abs()));
```

Reply:
As far as I know, this doesn't actually work because `OffsetDateTime::saturating_sub` takes a `Duration` but both arguments are `OffsetDateTime`. The only way to get `dt - dt` to produce a `Duration` without using `Sub` seems to be either reimplementing the logic within `Sub` or just working with "instants" (i.e. converting the dt to nanoseconds since the UNIX epoch) and doing the subtraction that way. 

Another consideration that might make it easier to work with is whether nanosecond precision is necessary, since by using the regular second precision `unix_timestamp` might be easier than dealing with an `i128`.